vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_authorization_unittest.cc

```diff
@@ -0,0 +1,299 @@
+            std::make_unique<UpholdAuthorization>(mock_ledger_impl_.get())} {}
+};
+
+std::string GetAnonFundsNameSuffixGenerator(
+    const TestParamInfo<GetAnonFunds::ParamType>& info) {
+  const bool fetch_old_balance_enabled = std::get<0>(info.param);
+  const std::string& brave_wallet = std::get<1>(info.param);
+  const type::UrlResponse& balance_server_response = std::get<2>(info.param);
+
+  if (!fetch_old_balance_enabled) {
+    return "fetch_old_balance_disabled";
+  }
+
+  if (brave_wallet.empty()) {
+    return "brave_wallet_is_not_created";
+  }
+
+  if (brave_wallet.find(R"("payment_id": "")") != std::string::npos) {
+    return "brave_wallet_payment_id_is_empty";
+  }
+
+  if (balance_server_response.status_code != net::HttpStatusCode::HTTP_OK) {
+    return "balance_server_error";
+  }
+
+  if (balance_server_response.body.empty()) {
+    return "invalid_body_in_balance_server_response";
+  }
+
+  return "happy_path";
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    UpholdAuthorizationTest,
+    GetAnonFunds,
+    Values(
+        // "fetch_old_balance_disabled"
+        GetAnonFundsParamType{false, {}, {}, {type::Result::LEDGER_OK, 0.0}},
+        // "brave_wallet_is_not_created"
+        GetAnonFundsParamType{true, {}, {}, {type::Result::LEDGER_OK, 0.0}},
+        // "brave_wallet_payment_id_is_empty"
+        GetAnonFundsParamType{
+            true,
+            R"({ "payment_id": "", "recovery_seed": "OG2zYotDSeZ81qLtr/uq5k/GC6WE5/7BclT1lHi4l+w=" })",
+            {},
+            {type::Result::LEDGER_ERROR, {}}},
+        // "balance_server_error"
+        GetAnonFundsParamType{
+            true,
+            R"({ "payment_id": "f375da3c-c206-4f09-9422-665b8e5952db", "recovery_seed": "OG2zYotDSeZ81qLtr/uq5k/GC6WE5/7BclT1lHi4l+w=" })",
+            type::UrlResponse{{},
+                              {},
+                              net::HttpStatusCode::HTTP_SERVICE_UNAVAILABLE,
+                              {},
+                              {}},
+            {type::Result::LEDGER_ERROR, {}}},
+        // "invalid_body_in_balance_server_response"
+        GetAnonFundsParamType{
+            true,
+            R"({ "payment_id": "f375da3c-c206-4f09-9422-665b8e5952db", "recovery_seed": "OG2zYotDSeZ81qLtr/uq5k/GC6WE5/7BclT1lHi4l+w=" })",
+            type::UrlResponse{{}, {}, net::HttpStatusCode::HTTP_OK, {}, {}},
+            {type::Result::LEDGER_ERROR, 0.0}},
+        // "happy_path"
+        GetAnonFundsParamType{
+            true,
+            R"({ "payment_id": "f375da3c-c206-4f09-9422-665b8e5952db", "recovery_seed": "OG2zYotDSeZ81qLtr/uq5k/GC6WE5/7BclT1lHi4l+w=" })",
+            type::UrlResponse{
+                {},
+                {},
+                net::HttpStatusCode::HTTP_OK,
+                R"({ "total": 5.0, "spendable": 0.0, "confirmed": 5.0, "unconfirmed": 0.0 })",
+                {}},
+            {type::Result::LEDGER_OK, 5.0}}),
+    GetAnonFundsNameSuffixGenerator);
+
+TEST_P(GetAnonFunds, ) {

```

Comment:
Yeah, omitting the comma here is a syntax error, since `TEST_P` expects two arguments.

Reply:
OK, gotcha. It feels a little weird to have no test name when it's expecting one, it might be good to name it something like `AnonFundsFlows` or something but up to you!

Reply:
Sure thing! I just wanted to find a way to best mimic the naming convention generated by `TEST_F` (e.g. `TEST_F(UpholdUtilTest, GetFeeAddress) { /* ... */ }` results in `[x/y] UpholdUtilTest.GetFeeAddress (z ms)`).
Providing only the test suite name for `TEST_P` and a suffix generator to `INSTANTIATE_TEST_SUITE_P` results in a somewhat similar naming scheme (e.g. `[x/z] UpholdAuthorizationTest/GetAnonFunds.brave_wallet_payment_id_is_empty (z ms)`).

But yeah, we could add something like `branches` as the `test_name` argument of `TEST_P`. Then we would end up with something like this: `[x/y] UpholdAuthorizationTest/GetAnonFunds.paths/brave_wallet_payment_id_is_empty (z ms)`.
What do you think?

Reply:
Ah OK, I see where you're going with your original idea! I guess I wouldn't worry too much about making the test name mimic those that use `TEST_F`. It will be kind of hard to do that anyway, since these tests have that extra "instantiation" parameter.

To me, the way this would look in the output should be something like this:

```
UpholdAuthorization/GetAnonFundsTest.AnonFundsFlow/EmptyPaymentId
UpholdAuthorization/GetAnonFundsTest.AnonFundsFlow/InvalidBody
...
```

i.e., `InstantiationName/TestSuiteName.TestName/Suffix`

Reply:
Resolved in e7a597ba21dd2e98bb46cc6f8791c33c848e6525.